### ISSUE TITLE
Amend the JDC election guidelines in Jupyter governance

### ISSUE THREAD

**Issue by afshin** [jupyter/distinguished-contributors#3]
## Synopsis
We should amend the Jupyter governance document that describes the Jupyter Distinguished Contributors (JDC) election process to include a lower bound as well as a higher bound of new members in each election.

There are several strategies we could propose but they each have drawbacks. In order to not color the conversation before it's begun, this issue simply lays out the question without offering a solution.

## Background

The [Jupyter governance document describing the JDC body](https://github.com/jupyter/governance/blob/master/distinguished_contributors.md#selecting-distinguished-contributors-members) sets out the process of selecting new Distinguished Contributors as follows:

> ### Selecting Distinguished Contributors Members
> 
> Distinguished Contributors are added to this body on an annual basis, through a voting process carried out by the pre-existing body of Distinguished Contributors. The number of candidates chosen each year will be limited to 10 new members. Nominations for election can be submitted by any Distinguished Contributor to be reviewed by a committee of peer reviewers chosen from the Distinguished Contributors. This committee is responsible for managing the private voting process and reviewing nominees to ensure that they meet the minimum criteria for Distinguished Contributor status. After nominees are finalized, the Distinguished Contributors will vote in a ranked preference process.

**Comment by Zsailer** [comment]
Opening the discussion here... 

Thinking about a lower bound raises a two questions in my mind.

*1) If we get fewer nominations than the lower bound, do we skip that year's JDC election or reopen the nomination period?*

Skipping a year because we get too few nominees might burn someone who deserves JDC. This could have major implications on that individual's career. On the other hand, reopening the nomination period opens the door for people to procrastinate.

The issue I see here current is that JDCs might be incentivized to find/nominate anyone who meets the minimum requirement (even if they're not enthusiastic about that nomination) to bring the count above the lower bound. This could dilute the impact of the JDC title. 

*2) If the number of nominations falls between the upper and lower bound, do we still hold a vote? What prevents us from accepting all nominees, other than not meeting the minimum criteria (e.g. a person doesn't meet the 2 year contribution period, etc.)?*

Voting makes sense when we have more than 10 nominees to rank them. When there is fewer than 10 nominees, however, this document seems to imply that all nominees are accepted if they meet the minimum criteria.

The issue I see here is that we lose an *implicit* vetting of nominees that happens when we rank.

While I believe we should err on the side of crediting more people, we may want to explore expanding the voting process to look beyond the minimum criteria.

**Comment by SylvainCorlay** [comment]
> Voting makes sense when we have more than 10 nominees to rank them. When there is fewer than 10 nominees, however, this document seems to imply that all nominees are accepted if they meet the minimum criteria.

I don't think that nominees should automatically be elected if there are not enough of them to reach the threshold. Having some kind of vote is probably preferable.

**Comment by choldgraf** [comment]
# Proposal

Add a second part to the voting process. Ask each voting member a question *in addition* to the ranked choice vote:

"Do you approve that this person be a Distinguished Contributor?"

optional answers are "yes", "no", and "abstain".

Any person will gain the DC title if they pass these conditions:

1. They receive >= a 66% percentage of `yes_votes / (yes_votes + no_votes)`.
1. They pass the ranked-choice vote.

If there are fewer nominations than max slots, then everybody automatically passes the ranked choice vote.

_Note: this isn't a binding proposal, just a straw-poll to see if people like it_

**Comment by SylvainCorlay** [comment]
Yes, that sounds great. What I think will happen is that it will be "yes" for everyone most of the time, but having some safeguard mechanism is good, to prevent the unlikely event that a completely random person gets nominated and elected just because there aren't enough nominees.

**Comment by Zsailer** [comment]
> This would be a simple "yes", "no", "abstain" to the question "Do you approve that this person be a Distinguished Contributor?".

I had this exact idea in mind. 

Though, to be more explicit, the fraction for approval should be `yes / (yes + no)`. We ignore abstentions. (Otherwise, as the JDC list grows larger, the fraction of "yes" votes becomes harder to obtain. Some JDCs might not participate—i.e. implicit "abstain"—or a nominee works on a single wing of the Jupyter community and maybe doesn't know many of the people on the current JDC list).

**Comment by choldgraf** [comment]
@Zsailer good point - I've updated [my comment above](https://github.com/jupyter/distinguished-contributors/issues/3#issuecomment-788165202) to be structured a bit more precisely, in the hopes that this makes it easier to comment and iterate. I also added your suggestion in there. I decided to go with 66% supermajority because I think that it would be a major red flag if half of the voting members explicitly did *not* want someone to be a distinguished contributor.

**Comment by damianavila** [comment]
+1 on the dual condition presented above.

**Comment by afshin** [comment]
This proposal sounds like a good idea to me. Strictly speaking this sounds compatible with the governance as it is written, as this describes a "rank preference process". If this proposal, or one in this vein, is the one we agree is best, should we submit a PR to governance or should we simply adopt this as an implementation detail of how we conduct a ranked preference process?

**Comment by Zsailer** [comment]
> should we submit a PR to governance or should we simply adopt this as an implementation detail of how we conduct a ranked preference process?

I think leaving it as an implementation detail is fine, especially since it's loosely documented here in this issue. 

I think we still need to answer the question: *what happens if we don't reach the lower bound number of nominees?*

I'm inclined to say that we skip electing JDC for a year and try again the next year. I think **3 people** is a good minimum bound (no particular reason, just looks like a good minimum to me).

**Comment by choldgraf** [comment]
What do people perceive as the downside to just having 1 nominee?

**Comment by afshin** [comment]
I think having a single JDC award one year would be fine.

**Comment by fperez** [comment]
It seems we never finalized this discussion - fwiw I think the above proposal by @choldgraf, with the fine-tuning from later feedback from @Zsailer et al., is solid.

I'd like to propose that for _future votes_ (so as to not modify the currently open one), we adopt the above model, so there's a chance to express not only ranking, but potentially disapproval if that exists (and as a safeguard too, as discussed above).

**Comment by afshin** [comment]
I imagined this happening in two phases: the CIVS ranked preference voting
and then a sanity check in a second round (mostly because I didn't see a
way to do this in one stage within CIVS). That does open the possibility of
adding that second step during this particular vote since we had that
conversation last time with the intention that its outcome would apply to
this current vote. I wouldn't argue for it too strenuously, I think
Fernando's suggestion that we wait for the next vote is also sound, but I
don't think it would harm the integrity of this particular election if we
apply a second stage to it.

**Comment by afshin** [comment]
I imagined this happening in two phases: the CIVS ranked preference voting
and then a sanity check in a second round (mostly because I didn't see a
way to do this in one stage within CIVS). That does open the possibility of
adding that second step during this cycle, since we had that conversation
last time with the intention that its outcome would apply to this current
vote.

I wouldn't argue for it too strenuously, I think Fernando's suggestion that
we wait for the next vote is also sound, but I don't think it would harm
the integrity of this particular election if we apply a second stage to it.

**Comment by meeseeksmachine** [comment]
This issue has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/governance-office-hours-meeting-minutes/1480/176
